src/lexer.mll

```diff
@@ -271,8 +271,82 @@ let raw_markup_target =
 let language_tag_char =
   ['a'-'z' 'A'-'Z' '0'-'9' '_' '-' ]
 
+rule reference_paren_content input start depth_paren depth_curly buffer = parse
+  | '('
+    {
+      Buffer.add_string buffer (Lexing.lexeme lexbuf) ;
+       reference_paren_content input start (depth_paren + 1) depth_curly buffer
+        lexbuf }
+  | '{'
+    {
+      Buffer.add_string buffer (Lexing.lexeme lexbuf) ;
+       reference_paren_content input start depth_paren (depth_curly + 1) buffer
+        lexbuf }
+  | ')'
+    {
+      Buffer.add_string buffer (Lexing.lexeme lexbuf) ;
+      if depth_paren = 0 then ()
+      else
+        ( reference_paren_content input start (depth_paren - 1) depth_curly
+            buffer lexbuf ) }
+  | '}'
+    {
+      Buffer.add_string buffer (Lexing.lexeme lexbuf) ;
+      if depth_curly = 0 then
+        warning
+          input
+          ~start_offset:(Lexing.lexeme_end lexbuf)
+          (Parse_error.not_allowed
+            ~what:"'}' (end of reference)"
+            ~in_what:(
+              Printf.sprintf "'%s' (custom operator)"
+                (Buffer.sub buffer 0 ((Buffer.length buffer) - 1))))
```

Comment:
I love this warning :)
Is this just `Buffer.contents buffer` ?

Reply:
No, I remove the last char of the buffer, that is always `}` in this situation

Reply:
Seems like the `}` can be added to the buffer only in the `else` branch, removing this weird dependency.